SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -124,6 +124,26 @@ void USpatialReceiver::OnAddEntity(const Worker_AddEntityOp& Op)
 	UE_LOG(LogSpatialReceiver, Verbose, TEXT("AddEntity: %lld"), Op.entity_id);
 }
 
+void USpatialReceiver::RemoveRedundantRemoveComponentOps(const Worker_AddComponentOp& Op)
+{
+	int32 RemovedOps = 0;
+	for (int32 i = QueuedRemoveComponentOps.Num() - 1; i >= 0; --i)
+	{
+		const Worker_RemoveComponentOp& RemoveComponentOp = QueuedRemoveComponentOps[i];
+		if (RemoveComponentOp.entity_id == Op.entity_id &&
+			RemoveComponentOp.component_id == Op.data.component_id)
+		{
+			QueuedRemoveComponentOps.RemoveAt(i);
+			RemovedOps++;
+		}
+	}
+
+	if (RemovedOps >= 2)
+	{
+		UE_LOG(LogSpatialReceiver, Warning, TEXT("Removing more than one RemoveComponentOp. Entity %d, Component %d"), Op.entity_id, Op.data.component_id);
```

Comment:
Is this a warning for debugging something right now?

Reply:
Not really. Valentyn suggested adding this warning as we did not think this case was valid.
It appears we are hitting this case which we talked a bit about yesterday.

Reply:
It might be that losing authority over a component that doesn't exist (so in the ACL but not on the entity) then removing the entity would generate it. I don't remember if the entity remove will cause the worker sdk to generate a remove component op for a component that doesn't exist.
I think the contract we get is actually just that adds and removes might be spurious, but won't cause inconsistencies.
It might be worth checking though as it's possible things rely on the details (although spatial view would also break that).

Reply:
I wouldn't expect that to be common but the way we handle dynamic components means that will definitely happen sometimes.

Reply:
Unless you mean not valid in the gdk because of hard handover of authority. Except we don't do that around possession and we don't do that for non generated components so I'm not 100% convinced it's definitely the case.

If we think we are getting it incorrectly then we should probably test it though.